### PR TITLE
Optimize storage writing

### PR DESCRIPTION
**Pull request by tolak**
This PR aims to optimize the storage API to reduce the count of access to the Internet because when converting deposit data fetched from smart contract to task data, there is a lot of HTTP requests will be sent, which will lead to contract revert (time limit). Now storage client only contains 4 APIs, which is:
- read_storage: read storage item according to the given key
- alloc_storage: create storage item
- update_storage: update storage item

### CODE REVIEW

File: contracts/index_executor/src/task.rs [hunk 1]
```diff
@@ -191,20 +203,32 @@ impl Task {
...
     pub fn destroy(&mut self, client: &StorageClient) -> Result<(), &'static str> {
-        let mut free_accounts = client.lookup_free_accounts();
-        let mut pending_tasks = client.lookup_pending_tasks();
-
-        if client.lookup_task(&self.id).is_some() {
+        let (mut free_accounts, free_accounts_doc) = client
+            .read_storage::<Vec<[u8; 32]>>(b"free_accounts")?
+            .ok_or("StorageNotConfigured")?;
+        let (mut pending_tasks, pending_tasks_doc) = client
+            .read_storage::<Vec<TaskId>>(b"free_accounts")?
```
Comment (h4x3rotab) [inline contracts/index_executor/src/task.rs:210]: Should be pending tasks?

Comment (tolak) [reply contracts/index_executor/src/task.rs:210]: Already fixed in following PR

File: contracts/index_executor/src/lib.rs [hunk 2]
```diff
@@ -346,61 +355,66 @@ mod index_executor {
...
-                    _ => {
-                        pink_extension::info!(
-                            "Task execution has not finished yet, update data to remote storage: {:?}",
-                            hex::encode(task.id)
-                        );
-                        client
-                            .put(task.id.as_ref(), &task.encode())
-                            .map_err(|_| Error::FailedToUploadTask)?;
-                        continue;
+            if let Some((ids, _)) = client
```
[the remark this answers is not in the document]
Comment (tolak) [reply contracts/index_executor/src/lib.rs:357]: It will return immediately if `read` returns None

**Review by h4x3rotab — APPROVED** [review]
LGTM

**Pull request merged**